### PR TITLE
Add option to logger to log thread names

### PR DESCRIPTION
**Pull request by Preston-Sundar**
Per issue firecracker-microvm#2117, added optional parameter to
allow logging of the current thread name in the prefix.
Added a rust unit test that spawns  a named thread and logs i
ts name in the prefix. Added python integration tests that
create logs with new show_thread_name parameter.

Signed-off-by: sundar.preston.789@gmail.com <sundar.preston.789@gmail.com>

Co-authored-by: Kevin Guo <kev.guo123@gmail.com>
Co-authored-by: Yichen Zhang <jakezhang2020@gmail.com>

## Reason for This PR

Addresses issue #2117 

## Description of Changes

* Modified `src/logger/src/logger.rs` to optionally allow appending the thread name to the log prefix. Enabled by setting `show_thread_name` to true. Also added unit test that creates a thread with custom name and checks the log's prefix for that name.
* Modified `src/vmm/src/vmm_config/logger.rs` so that `LoggerConfig`'s signature matches changes made to the logger.
* Added argument `show-thread-name` to the argument parser in `src/firecracker/src/main.rs`.
* Minor changes made to `tests/framework/resources.py`, `src/vmm/src/rpc_interface.rs`, and `src/api_server/src/request/logger.rs` to correctly call the logger with the new parameter.
* Added documentation changes in `docs/device-api.md`, `docs/logger.md`, and `src/api_server/swagger/firecracker.yaml` that correspond to the new parameter. 

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

**Comment by berciuliviu** [comment]
Hey there!

I have investigated your solution and I have discussed with the team the approach provided. We decided that **logging the thread name** could by done by **default**, thus avoiding an additional CL parameter and the added complexity it brings. This would simplify the solution by a lot, reducing it only to the logic of logging the thread name and the afferent tests.

File: tests/integration_tests/functional/test_logging.py [hunk 1]
```diff
@@ -89,6 +93,24 @@ def test_no_level_logs(test_microvm_with_ssh):
...
+
+
+def test_show_only_thread_name(test_microvm_with_ssh):
+    """Check if logs contain thread name when origin and level are disabled."""
+    _test_log_config(
+        microvm=test_microvm_with_ssh,
+        show_level=False,
+        show_origin=False,
+        show_thread_name=True
+    )
```
Comment (berciuliviu) [inline tests/integration_tests/functional/test_logging.py:111]: Pytests & modifications on this part could be separated in another commit.

**Comment by Preston-Sundar** [comment]
Hey @berciuliviu! Thank you for reviewing this, 

I'll modify the PR to log the thread name by default and remove the parameter logic. I'll also split the Pytests into a separate commit as well. 

Thanks!

**Comment by berciuliviu** [comment]
Hello Sundar,

Are there any updates on this work?

**Comment by Preston-Sundar** [comment]
Created new PR #2352 because of accidentally messing up `git rebase`.  😢.
